### PR TITLE
resolved_ts: update leader info after conf change in raftstore v2

### PR DESCRIPTION
**Pull request by 3pointer**
### What is changed and how it works?

Issue Number: ref https://github.com/tikv/tikv/issues/15192

What's Changed:
update leader info for the leader peer after conf change.

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with None.
If this PR will be picked to release branch, then a release note is probably required.
```

### CODE REVIEW

File: components/raftstore-v2/src/operation/command/admin/conf_change.rs [hunk 1]
```diff
@@ -196,6 +196,9 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             .lock()
             .unwrap()
             .set_region(self.region(), true, &self.logger);
+        // Update leader's peer list after conf change.
+        self.read_progress()
```
Comment (overvenus) [inline components/raftstore-v2/src/operation/command/admin/conf_change.rs:200]: Could you check other place where `set_region` is called but not does not call `update_leader_info`?

Comment (3pointer) [reply components/raftstore-v2/src/operation/command/admin/conf_change.rs:200]: I've checked `on_apply_res_split` and `on_apply_res_commit_merge`. both have called `update_leader_info` in `set_region`. can we directly call set_region here?

**Review by overvenus — APPROVED** [review]

**Review by tabokie — APPROVED** [review]

**Pull request merged**